### PR TITLE
Add pre-commit for CODEOWNERS

### PR DESCRIPTION
**Pull request by edmundmiller**
- **style: Copy pre-commit config from tools**
- **ci: Add codeowners-validator to pre-commit**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.2.2
+    hooks:
+      - id: ruff # linter
+        args: [--fix, --exit-non-zero-on-fix] # sort imports and fix
+      - id: ruff-format # formatter
+
```
Comment (mashehu) [inline .pre-commit-config.yaml:8]: no need for ruff in configs, I think

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    rev: "v1.8.0"
+    hooks:
+      - id: mypy
+        additional_dependencies:
+          - types-PyYAML
+          - types-requests
+          - types-jsonschema
+          - types-Markdown
+          - types-setuptools
+
```
Comment (mashehu) [inline .pre-commit-config.yaml:30]: same with mypy

File: .pre-commit-config.yaml (hunk 1 above)
Comment (edmundmiller) [reply .pre-commit-config.yaml:8]: There's a python script in `bin` though

Comment (mashehu) [reply .pre-commit-config.yaml:8]: ah nevermind then

**Review by mashehu — APPROVED** [review]

**Pull request merged**